src/otx/engine/engine.py

```diff
@@ -323,9 +323,42 @@ def predict(
             return_predictions=return_predictions,
         )
 
-    def export(self, *args, **kwargs) -> None:
-        """Export the trained model to OpenVINO Intermediate Representation (IR) or ONNX formats."""
-        raise NotImplementedError
+    def export(self, checkpoint: str | Path | None = None, export_config: ExportConfig | None = None, **kwargs) -> Path:
+        """
+        Export the trained model to OpenVINO Intermediate Representation (IR) or ONNX formats.
+
+        Args:
+            checkpoint (str | Path | None, optional): Checkpoint to export. Defaults to None.
+            export_config (ExportConfig | None, optional): Config that allows to set export format and precision. Defaults to None.
+
+        Returns:
+            Path: Path to the exported model.
+        """
+        ckpt_path = str(checkpoint) if checkpoint is not None else self.checkpoint
+        if export_config is None:
+            export_config = ExportConfig()
+
+        if ckpt_path is not None:
+            self.model.eval()
+            # self.model.label_info = self.datamodule.meta_info this doesn't work for some models yet
+            lit_module = self._build_lightning_module(
+                model=self.model,
+                optimizer=self.optimizer,
+                scheduler=self.scheduler,
+            )
+            lit_module.meta_info = self.datamodule.meta_info
+
+            loaded_checkpoint = torch.load(ckpt_path)
+            lit_module.load_state_dict(loaded_checkpoint["state_dict"])
```

Comment:
```suggestion
            lit_module = self._build_lightning_module(
                model=self.model,
                optimizer=self.optimizer,
                scheduler=self.scheduler,
            )
            
            loaded_checkpoint = torch.load(ckpt_path)
            
            # We need to use the information from the checkpoint when we want to export
            # In this phase, Datamodule doesn't have any real information. It also needs to be fixed.
            # At the export phase, we don't need to any input for the data.
            
            lit_module.meta_info = loaded_checkpoint["meta_info"]
            lit_module.load_state_dict(loaded_checkpoint["state_dict"])
```

Reply:
Unfortunately, `loaded_checkpoint` doesn`t contain `meta_info` key. Also that wouldn't solve problem of missing `model.label_info`

Reply:
It's in loaded_checkpoint ['state_dict']['meta_info'].

Reply:
Thanks, that field exists, but label_info of the model is still outdated even if we run `lit_module.meta_info = loaded_checkpoint['state_dict']["meta_info"]`